### ISSUE TITLE
[Feature] Multi-Teacher distiller

### ISSUE THREAD

**Issue by CatLoves** [open-mmlab/mmrazor#597]
### Describe the feature

A module like the SingleTeacherDistill which supports two or more teacher models and a single student, for an example, the student model is a multi-task model consisting of 2D object detection, 3D object detection and semantic segmentation tasks, the teachers are:
(1) A strong 2D object detection model;
(2) A strong 3D object detection model;
(3) A strong semantic segmentation model;
For the student model, all tasks share the same backbone but have different neck/heads; 
For the teacher model, all tasks have the same feature/logits tensor shape as the student's corresponding task;
During training process, we expect MultiTeacherDistill module to calculate distill loss from 3 tasks, for each task the loss between student 
feature/logits tensor and teacher feature/logits tensor is computed, then distill loss is calculated by weighting the 3 losses.

### Motivation

A clear and concise description of the motivation of the feature.
Ex1. It is inconvenient when the sudent is multi-task model and we have two or more teacher models for supervision.

### Related resources

If there is an official code release or third-party implementation, please also provide the information here, which would be very helpful.

### Additional context

If you would like to implement the feature and create a PR, please leave a comment here and that would be much appreciated.
